mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -137,6 +138,17 @@ class KMeansModel private[ml] (
   @Since("1.6.0")
   override def write: MLWriter = new KMeansModel.KMeansModelWriter(this)
 
+  override def hashCode(): Int = {
+    (Array(this.getClass, uid) ++ clusterCenters)
```

Comment:
@sethah Ooops I missed this one before. My apologies. I agree with you. I'll redesign the equality checking.